django_mongodb/query_utils.py

```diff
@@ -27,13 +27,21 @@ def process_lhs(node, compiler, connection):
 
 def process_rhs(node, compiler, connection):
     rhs = node.rhs
+    db_rhs = getattr(rhs, "_db", None)
+    if db_rhs is not None and db_rhs != connection.alias:
```

Comment:
I see this logic from `django.db.models.lookups.In`. Can it go in `django_mongodb.lookups._in` so all other lookups don't have the overhead?